openslides_backend/restrictions/restrictions.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Restrictions handler. It is the concret implementation of Restrictions interface.
+    """
+
+    def handle_request(
+        self, payload: Payload, logging: LoggingModule, services: Services,
+    ) -> RestrictionResponse:
+        """
+        Takes payload and user id and handles this request by validating and
+        parsing all fields. The fields are restricted according to the user id.
+        """
+        self.logging = logging
+        self.logger = logging.getLogger(__name__)
+        self.permission = services.permission
+        self.database = services.database
+
+        # Validate payload of request
+        try:
+            self.validate(payload)
+        except JsonSchemaException as exception:
+            raise RestrictionException(exception.message)
+
+        # Parse restrictions and creates response
+        response = self.parse_restrictions(payload)
+        self.logger.debug("Request was successful. Send response now.")
+        return response
+
+    def validate(self, payload: Payload) -> None:
+        """
+        Validates restrictions requests sent by client. Raises JsonSchemaException if
+        input is invalid.
+        """
+        self.logger.debug("Validate restrictions request.")
+        payload_schema(payload)
+
+    def parse_restrictions(self, payload: Payload) -> RestrictionResponse:
+        """
+        Parses restrictions request send by client. Raises RestrictionException
+        if something went wrong.
+        """
+        # permissions = self.permission().get_all(self.user_id)
+        self.logger.debug(
+            f"Restrictions map contains the following restrictions: {restrictions_map}."
+        )
+        response = []
+        for restriction_blob in payload:
+            # user_id = restriction_blob["user_id"]
+            blob_response = {}
+            for key in restriction_blob["fqfields"]:
+                collection_str, id, field = key.split(KEYSEPARATOR)
+                fqfield = FullQualifiedField(Collection(collection_str), int(id), field)
+                # restriction = restrictions_map.get(fqfield.collection)
+                # if restriction is None:
+                #     raise RestrictionException(
+                #         f"Restriction for {str(fqfield.collection)} does not exist."
+                #     )
+                # self.logger.debug(f"Perform restriction for {fqfield}.")
+                # blob_response[fqfield] = restriction(permissions, self.database()).perform(
+                #     fqfield, self.user_id
+                # )
+                blob_response[fqfield] = self.dummy_restriction()
+            response.append(blob_response)
+        self.logger.debug("Restricted data ready.")
+        return response
+
+    def dummy_restriction(self) -> Any:
+        # Just a dummy restrictor, that restricts nothing but always send the same message.
+        return "This is a restricted field content made by dummy restrictor."
```

Comment:
The autoupdate service only send values to the client when the value as changed. Thins this dummy restricter always returns the same data, the autoupdate sends no new data to a (test-) client.

Could you add some variants in the output? For example the current time, a random number or a ascending counter.

Reply:
Of course. The dummy client will send whatever you want (and code in your next pull request ;-) ).